semantic/src/main/kotlin/tools/samt/semantic/SemanticModelAnnotationProcessor.kt

```diff
@@ -0,0 +1,117 @@
+                        context.error {
+                            message("Unknown annotation @${name}")
+                            highlight("invalid annotation", annotation.location)
+                        }
+                    }
+                }
+            }
+        }
+        return UserMetadata(descriptions, deprecations)
+    }
+
+    private fun Package.getAnnotatedElements(): List<Annotated> = buildList {
+        this@getAnnotatedElements.allSubPackages.forEach {
+            addAll(it.records)
+            it.records.flatMapTo(this, RecordType::fields)
+            addAll(it.enums)
+            addAll(it.aliases)
+            addAll(it.services)
+            val operations = it.services.flatMap(ServiceType::operations)
+            addAll(operations)
+            operations.flatMapTo(this, ServiceType.Operation::parameters)
+        }
+    }
+
+    private fun getDescription(annotation: AnnotationNode): String {
+        check(annotation.name.name == "Description")
+        val arguments = annotation.arguments
+        val context = controller.getOrCreateContext(annotation.location.source)
+        if (arguments.isEmpty()) {
+            context.error {
+                message("Missing argument for @Description")
+                highlight("invalid annotation", annotation.location)
+            }
+            return ""
+        }
+        if (arguments.size > 1) {
+            val errorLocation = annotation.location.copy(
+                start = arguments[1].location.start,
+                end = arguments.last().location.end
+            )
+            context.error {
+                message("@Description expects exactly one string argument")
+                highlight("extraneous arguments", errorLocation)
+            }
+        }
+        return when (val description = arguments.first()) {
+            is StringNode -> description.value
+            else -> {
+                context.error {
+                    message("Argument for @Description must be a string")
+                    highlight("invalid argument type", description.location)
+                }
+                ""
+            }
+        }
+    }
+
+    private fun getDeprecation(annotation: AnnotationNode): UserMetadata.Deprecation {
+        check(annotation.name.name == "Deprecated")
+        val context = controller.getOrCreateContext(annotation.location.source)
+        val description = annotation.arguments.firstOrNull()
+        if (description != null && description !is StringNode) {
+            context.error {
+                message("Argument for @Deprecated must be a string")
+                highlight("invalid argument type", description.location)
+            }
+        }
+        if (annotation.arguments.size > 1) {
+            val errorLocation = annotation.location.copy(
+                start = annotation.arguments[1].location.start,
+                end = annotation.arguments.last().location.end
+            )
```

Comment:
Does a `copy` really make sense here?

Reply:
How about `annotation.arguments[1].location.copy(end = annotation.arguments.last().location.end)`

Reply:
that makes more sense, I like it👍